exporter/signalfxexporter/dpclient.go

```diff
@@ -48,37 +51,81 @@ type sfxDPClient struct {
 
 func (s *sfxDPClient) pushMetricsData(
 	ctx context.Context,
-	md consumerdata.MetricsData,
+	md pdata.Metrics,
 ) (droppedTimeSeries int, err error) {
-	accessToken := s.retrieveAccessToken(md)
+	metricsData := pdatautil.MetricsToMetricsData(md)
+	var numDroppedTimeseries int
+	var errs []error
+
+	var currentToken string
+	var batchStartIdx int
+	for i := range metricsData {
+		metricToken := s.retrieveAccessToken(metricsData[i])
+		if currentToken != metricToken {
+			if batchStartIdx < i {
+				// TODO:
+				// 1) To leverage the queued retry mechanism available in new exporters,
+				// 	  look for error cases that are non-permanent and track the list of metric
+				//    data that needs to be retried. This will be done once partial retry support
+				//    is added for metrics type.
+				// 2) Also consider invoking in a goroutine to some concurrency going, in case
+				//    there are a lot of tokens.
+				droppedCount, err := s.pushMetricsDataForToken(metricsData[batchStartIdx:i], currentToken)
```

Comment:
Since there's this check `batchStartIdx < i` before a push, I think `currentToken` will have the current value for a push. I mean the `currentToken` value is set when the first element with that token is encountered and at a later point when a new token is seen, it'll use the `currentToken` before setting the new value. Can you think of a case where this wouldn't work as intended?

Reply:
Oh yes, you are right!